File: Java_project/UI_codes/SendMail.java
package UI_codes;
import java.util.Properties;
import javax.mail.*;
import javax.mail.internet.*;
import java.net.*;
import java.util.*;
import java.util.Base64;
import java.text.SimpleDateFormat;
import java.nio.file.*;
import java.sql.*;
import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import javax.xml.bind.DatatypeConverter;
public class SendMail {
	public static void contactMail(String inString, Data s) throws Exception{
		String name = URLDecoder.decode(s.extract("name=(.*)&mailid",inString),"UTF-8");
		String emailid = URLDecoder.decode(s.extract("mailid=(.*)&body",inString),"UTF-8");
		String body = URLDecoder.decode(s.extract("body=(.*)&Send=",inString),"UTF-8");	 

		  Properties props = new Properties();
		  props.put("mail.smtp.host", "smtp.gmail.com");
		  props.put("mail.smtp.socketFactory.port", "465");
		  props.put("mail.smtp.socketFactory.class",
				"javax.net.ssl.SSLSocketFactory");
		  props.put("mail.smtp.auth", "true");
		  props.put("mail.smtp.port", "465");
		 
		  Session session = Session.getDefaultInstance(props,
			   new javax.mail.Authenticator() {
			   protected javax.mail.PasswordAuthentication getPasswordAuthentication() {
			  	 return new javax.mail.PasswordAuthentication("a.hs771994@gmail.com","ClarkKent");
			   }
		  });	 

		  try {
			   MimeMessage message = new MimeMessage(session);
			   MimeMessage messagedev = new MimeMessage(session);
			   messagedev.setFrom(new InternetAddress("a.hs771994@gmail.com"));
			   messagedev.addRecipient(Message.RecipientType.TO,new InternetAddress("a.hs771994@gmail.com"));
			   messagedev.setSubject("From Application");
			   messagedev.setText("name : "+name+"\nmail: "+emailid+"\nbody: \n"+body);
			   Transport.send(messagedev);
			   message.setFrom(new InternetAddress("a.hs771994@gmail.com"));
			   message.addRecipient(Message.RecipientType.TO,new InternetAddress(emailid));
			   message.setSubject("Concerning Your Feedback");
			   message.setText("Dear ,\n\tYour feedback on our application has been received with gratitude and is being reviewed by our developers. We promise to respond soon on this matter. \nThanking you,\n\nregards,\nJavaApp\nNOTE: This is an auto-generated email. Kindly do not reply.");
			   Transport.send(message);
		  } catch (MessagingException e) {throw new RuntimeException(e);}
	 
	 }
}

File: ruby_project/UI_scripts/sendmail.rb
class Send
	def self.mail(response)
		options = { 
		   :address              => "smtp.gmail.com",
		   :port                 => 587,
		   :user_name            => 'a.hs771994',
		   :password             => 'ClarkKent',
		   :authentication       => 'plain',
		   :enable_starttls_auto => true  }

		Mail.defaults do
		  delivery_method :smtp, options
		end
		name = CGI.unescape(response.scan(/name=(.*)&mailid/).flatten.pop)
		mailid = CGI.unescape(response.scan(/mailid=(.*)&body/).flatten.pop)
		body = CGI.unescape(response.scan(/body=(.*)&Send/).flatten.pop)
		Mail.deliver do
			to 'a.hs771994@gmail.com'
			from 'a.hs771994@gmail.com'
			subject 'from website!!'
			body "Name :#{name}\nMail - id : #{mailid}\nMessage : #{body}"
		end
		Mail.deliver do
			to mailid
			from 'a.hs771994@gmail.com'
			subject 'Concerning your feedback'
			body "Dear #{name},\n\tYour feedback on our application has been received with gratitude and is being reviewed by our developers. We promise to respond soon on this matter. \nThanking you,\n\nregards,\nTaskApp\nNOTE: This is an auto-generated email. Kindly do not reply."
		end
	end
	
end
